src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -789,8 +794,10 @@ namespace winrt::TerminalApp::implementation
     // Arguments:
     // - term: The newly created TermControl to connect the events for
     // - hostingTab: The Tab that's hosting this TermControl instance
-    void TerminalPage::_RegisterTerminalEvents(TermControl term, std::shared_ptr<Tab> hostingTab)
+    void TerminalPage::_RegisterTerminalEvents(TermControl term, winrt::TerminalApp::Tab hostingTab)
```

Comment:
Could this just take the implementation type directly (or via a reference or whatever), so we don't need to call `get_self` in the body of this method, after the callers already have a `com_ptr` for the implementation anyways?

Reply:
That makes sense, updated `_RegisterTerminalEvents` to take a `com_ptr` instead.